Resources/config/monolog.xml

```diff
@@ -40,5 +40,9 @@
         <service id="monolog.http_client" class="Symfony\Contracts\HttpClient\HttpClientInterface" public="false">
             <factory class="Symfony\Component\HttpClient\HttpClient" method="create" />
         </service>
+
+        <service id="monolog.handler_manager" class="Symfony\Bundle\MonologBundle\HandlerLifecycleManager" public="true">
+            <argument type="tagged_iterator" tag="monolog.handler" />
```

Comment:
Rather than using `type="tagged_iterator"`, you should use a compiler pass to create the IteratorArgument (or drop the tag entirely and manage everything in the DI extension as I don't think we need to support handlers added elsewhere). This way, you can use weak references in the IteratorArgument, which will skip services which haven't been instantiated yet

Reply:
I'm not sure I'm familiar enough with the Container to change this right now  but given https://github.com/symfony/monolog-bundle/pull/377#discussion_r531587372 I should probably pause on this PR now anyway?